include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -1745,18 +1764,16 @@ namespace internal
         fe_patch_values.get_cell()->index(),
         this->dof_handler);
 
-      const VectorType *vector = &((*vectors)[dof_cell->level()]);
+      const auto *vector = &(vectors[dof_cell->level()]);
 
       const unsigned int dofs_per_cell =
         this->dof_handler->get_fe(0).n_dofs_per_cell();
 
       std::vector<types::global_dof_index> dof_indices(dofs_per_cell);
       dof_cell->get_mg_dof_indices(dof_indices);
       std::vector<double> values(dofs_per_cell);
-      VectorHelper<VectorType>::extract(*vector,
-                                        dof_indices,
-                                        extract_component,
-                                        values);
+      VectorHelper<LinearAlgebra::distributed::BlockVector<ScalarType>>::
+        extract(*vector, dof_indices, extract_component, values);
```

Comment:
When this patch is in, *every* occurrence of the template argument of `VectorHelper` should be of `LA::distributed` type, right? The difference will only be whether we have blocked or non-blocked vectors. Can we simplify things in that case?

Reply:
Yes :+1: